src/main/scala/viper/silver/plugin/standard/termination/transformation/DecreasesCheck.scala

```diff
@@ -43,9 +43,15 @@ trait DecreasesCheck extends ProgramManager with ErrorReporter {
     val rCondition = requiredCondition.getOrElse(TrueLit()(errT = reTrafo))
     val gCondition = givenCondition.getOrElse(TrueLit()(errT = reTrafo))
 
-    val or = Or(Not(rCondition)(errT = reTrafo), gCondition)(errT = reTrafo)
-
-    Assert(or)(errT = errTrafo)
+    // Generates the statement "assert !rCondition || gCondition". If either rCondition or gCondition are the
+    // true literal, their respective disjuncts are simplified or even removed. This ensures that trivial assertions
+    // like "assert true" and "assert !true || true" are kept to a minimum.
+    (rCondition, gCondition) match {
+      case (_: TrueLit, _: TrueLit) => EmptyStmt
+      case (_: TrueLit, _) => Assert(gCondition)(errT = errTrafo)
+      case (_, _: TrueLit) => Assert(Not(rCondition)(errT = reTrafo))(errT = errTrafo)
```

Comment:
The third case looks wrong to me, should just be true, right? The rest looks right?
Also, I'm confused why we're not using implications everywhere (both in the old version and in the new version). !r || g is generally a weird way of writing r ==> g

Reply:
thanks @ArquintL and @marcoeilers, you are right. The third case was wrong. Should be addressed now. As for the use of implications, there is a comment a few lines before the change set that explains it:
```
    // When Implies is used, error reason would point to right side of implication (i.e. givenCondition),
    // for which no reason transformation can be defined. (this happens in carbon)
    // To avoid this and equivalent expression using Or is used.
```